src/main/java/com/minecolonies/coremod/blocks/BlockShingleSlab.java

```diff
@@ -0,0 +1,261 @@
+    {
+        return this.getDefaultState().withProperty(FACING, EnumFacing.HORIZONTALS[meta]);
+    }
+
+    /**
+     * @deprecated remove when minecraft invents something better.
+     */
+    @Deprecated
+    @Override
+    public AxisAlignedBB getBoundingBox(@NotNull final IBlockState state, @NotNull final IBlockAccess source, @NotNull final BlockPos pos)
+    {
+        return AABB_BOTTOM_HALF;
+    }
+
+    /**
+     * @deprecated remove when minecraft invents something better.
+     */
+    @Deprecated
+    @Override
+    public IBlockState getActualState(@NotNull final IBlockState state, @NotNull final IBlockAccess worldIn, @NotNull final BlockPos pos)
+    {
+        return getSlabShape(state, worldIn, pos);
+    }
+
+    /**
+     * Get the step shape of the slab
+     * @param state the state.
+     * @param world the world.
+     * @param position the position.
+     * @return the blockState to use.
+     */
+    private static IBlockState getSlabShape(@NotNull final IBlockState state, @NotNull final IBlockAccess world, @NotNull final BlockPos position)
+    {
+        final boolean[] connectors = new boolean[]{!(world.getBlockState(position.east()).getBlock() instanceof BlockShingleSlab),
+            !(world.getBlockState(position.west()).getBlock() instanceof BlockShingleSlab),
+            !(world.getBlockState(position.north()).getBlock() instanceof BlockShingleSlab),
+            !(world.getBlockState(position.south()).getBlock() instanceof BlockShingleSlab)};
+
+        int amount = 0;
+        for(final boolean check: connectors)
+        {
+            if(check)
+            {
+                amount++;
+            }
+        }
+
+        if(amount == NO_CONNECTIONS)
+        {
+            return state.withProperty(VARIANT, ShingleSlabType.TOP);
+        }
+        if(amount == ONE_CONNECTION)
+        {
+            if (connectors[0])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.ONE_WAY).withProperty(FACING, EnumFacing.SOUTH);
+            }
+            else if (connectors[1])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.ONE_WAY).withProperty(FACING, EnumFacing.NORTH);
+            }
+            else if (connectors[2])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.ONE_WAY).withProperty(FACING, EnumFacing.EAST);
+            }
+            else
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.ONE_WAY).withProperty(FACING, EnumFacing.WEST);
+            }
+        }
+        else if(amount == TWO_CONNECTIONS)
+        {
+            if (connectors[0] && connectors[1] && !connectors[2] && !connectors[3])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.TWO_WAY).withProperty(FACING, EnumFacing.EAST);
+            }
+            else if (!connectors[0] && !connectors[1] && connectors[2] && connectors[3])
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)